crates/blind_pool/src/pool.rs

```diff
@@ -557,4 +557,171 @@ mod tests {
 
         assert!(pool.is_empty());
     }
+
+    #[test]
+    fn trait_object_usage() {
+        use std::fmt::Display;
```

Comment:
Unclear why this `use` statement is scoped to this function. We generally do not do function-scoped `use` statements for no reason.